service/storage_service.cc

```diff
@@ -1922,7 +1922,7 @@ future<> storage_service::join_token_ring(sharded<db::system_distributed_keyspac
     }
 
     assert(_group0);
-    co_await _group0->finish_setup_after_join(*this, _qp, _migration_manager.local(), _raft_experimental_topology);
+    co_await _group0->finish_setup_after_join(*this, _qp, _migration_manager.local(), false);
```

Comment:
@piodul please review this commit (` storage_service: join_token_ring: fix finish_setup_after_join calls `)

Reply:
Looks good.

I suppose we didn't catch this earlier because that would require doing a rolling upgrade from a version which didn't have a group0 created yet (if we already joined then we won't create the `group0_state_machine` again), but the code after the change is obviously more correct.